Frontend/isbit/src/components/dataset-selection/DataSelection.vue

```diff
@@ -1,121 +1,283 @@
+                <option value="COMBO">COMBO</option>
+              </select> 
+            </div>
           </div>
-          <button @click="redirectToDetails(item.dataset)">Uppmärk Data</button>
         </div>
       </div>
     </div>
   </div>
 </template>
 
 <script lang="ts">
-import { defineComponent, ref, onMounted } from 'vue'
-import axios from 'axios'
+import { defineComponent, ref, onMounted } from 'vue';
+import axios from 'axios';
+
 export default defineComponent({
   name: 'DataDisplay',
   setup() {
-    const dataList = ref<any[]>([])
-    const error = ref<string | null>(null)
+    const dataList = ref<any[]>([]);
+    const availableDatasets = ref<any[]>([]);
+    const combinedDataList = ref<any[]>([]);
+    const error = ref<string | null>(null);
+    const loading = ref<boolean>(false); 
+
     const fetchData = async () => {
       try {
-        //TODO: add base url to const file
-        const response = await axios.get('http://localhost:8000/V1/dataset')
-        console.log('Data fetched:', response.data)
-        dataList.value = response.data.dataList
+        const filesResponse = await axios.get('http://localhost:8000/V1/dataset/files');
+        dataList.value = filesResponse.data.dataList;
+
+        const availableDataResponse = await axios.get('http://localhost:8000/V1/dataset');
+        availableDatasets.value = availableDataResponse.data.dataList;
+
+        combineData();
       } catch (err) {
-        error.value = 'Error fetching data'
-        console.error(err)
+        error.value = 'Error fetching data';
+        console.error(err);
       }
-    }
-    onMounted(fetchData)
+    };
+
+    const combineData = () => {
+      combinedDataList.value = dataList.value.map(file => {
+        const availableData = availableDatasets.value.find(item => item.dataset === file.dataset);
+        return {
+          ...file,
+          assignment: availableData ? availableData.assignment : null,
+          datatype: availableData ? availableData.datatype : null,
+          loading: false,
+          dimRedMethod: 'TSNE'
+        };
+      });
+    };
+
+    const isDataAvailable = (dataset: string): boolean => {
+      return availableDatasets.value.some(item => item.dataset === dataset);
+    };
+
     return {
-      dataList,
-      error
-    }
+      combinedDataList,
+      error,
+      isDataAvailable,
+      fetchData,
+      loading 
+    };
   },
   methods: {
     redirectToDetails(dataset: string) {
-      this.$router.push({ path: '/label', query: { dataset: dataset } })
+      this.$router.push({ path: '/label', query: { dataset: dataset } });
+    },
+    async runModel(file: string, index: number, dimRedMethod: string) {
+      try {
+        this.combinedDataList[index].loading = true;
+        //TOOD: fix model name
+        await axios.post('http://localhost:8000/V1/run_ml/?model_name=qaqc_main&file=' + file + '&dim_red_method=' + dimRedMethod);
```

Comment:
Is there a way to get the model name now or does it still need to be hard-coded?

Reply:
not possible right now, Leo made a backend issue for it

Reply:
alright, this is fine then